lxd/container.go

```diff
@@ -1657,6 +1662,75 @@ func autoCreateContainerSnapshots(ctx context.Context, d *Daemon, containers []c
 	return nil
 }
 
+func pruneExpiredContainerSnapshotsTask(d *Daemon) (task.Func, task.Schedule) {
+	f := func(ctx context.Context) {
+		opRun := func(op *operation) error {
+			return pruneExpiredContainerSnapshots(ctx, d)
+		}
+
+		op, err := operationCreate(d.cluster, "", operationClassTask, db.OperationSnapshotsExpire, nil, nil, opRun, nil, nil)
```

Comment:
We should do the same thing here as we do for backup expiry, so avoid running the whole thing if there is nothing to expire, saves us noisy logging. I can send a fix for that later.